File: lab2/Python (Zad 2.2-2.6, 2.8)/serialize_json.py
#
"""
serialize json
"""
import json

class SerializeJson:
    # metoda statyczna
    @staticmethod
    def run(deserializeddata, filelocation):
        print("\nlet's serialize something...")
        lst = []
        # TODO – do modyfikacji
        for dep in deserializeddata.data:
            lst.append({
                "Kod_TERYT": dep['Kod_TERYT'],
                "Województwo": dep['Województwo'],
                "Powiat": dep['Powiat'],
                "typ_JST": dep['typ_JST'],
                "nazwa_urzędu_JST": dep['nazwa_urzędu_JST'],
                "miejscowość": dep['miejscowość'],
                "telefon": dep['telefon'],
                "telefon_kierunkowy": dep['telefon kierunkowy']
            })
        jsontemp = {"departaments": lst}
        with open(filelocation, 'w', encoding='utf-8') as f:
            json.dump(jsontemp, f, ensure_ascii=False)
        print("all data has been serialized!")

    def run_xml(deserializeddata, filelocation):
        print("\nLet's serialize something...")
        lst = []
        for dep in deserializeddata['data']:
            lst.append({
                "Kod_TERYT": dep['Kod_TERYT'],
                "Województwo": dep['Województwo'],
                "Powiat": dep['Powiat'],
                "typ_JST": dep['typ_JST'],
                "nazwa_urzędu_JST": dep['nazwa_urzędu_JST'],
                "miejscowość": dep['miejscowość'],
                "telefon_kierunkowy": dep['telefon_kierunkowy']
            })
        jsontemp = {"JST": lst}
        with open(filelocation, 'w', encoding='utf-8') as f:
            json.dump(jsontemp, f, ensure_ascii=False)
        print("All data has been serialized!")

File: lab2/C# (Zad 2.7)/Plik zrodlowy/SerializeJson.cs
namespace IS_Lab2_JSON;

using System.Text.Encodings.Web;
using System.Text.Json;
using System.Text.Unicode;
using System.Text;

public class SerializeJson
{
    public static void Run(string path, string savePath)
    {
        Console.WriteLine("\nlet's serialize something...");
        DeserializeJson deserialized = new DeserializeJson(path);
        List<ShorterJST> shorterJsts = new List<ShorterJST>();
        foreach (var d in deserialized.data)
            shorterJsts.Add(shorterJSTMapper(d));

        var options = new JsonSerializerOptions
        {
            Encoder = JavaScriptEncoder.Create(UnicodeRanges.LatinExtendedA, UnicodeRanges.LatinExtendedB, UnicodeRanges.Latin1Supplement, UnicodeRanges.BasicLatin),
            WriteIndented = true
        };
        string json = JsonSerializer.Serialize<List<ShorterJST>>(shorterJsts, options);
        File.WriteAllText(savePath, json, Encoding.UTF8);
        Console.WriteLine("all data has been serialized!");
    }

    public static void Run(DeserializeJson deserialized, string savePath)
    {
        Console.WriteLine("let's serialize something");
        List<ShorterJST> shorterJsts = new List<ShorterJST>();
        foreach (var d in deserialized.data)
            shorterJsts.Add(shorterJSTMapper(d));

        string json = JsonSerializer.Serialize<List<ShorterJST>>(shorterJsts);
        File.WriteAllText(savePath, json);
        Console.WriteLine("all data has been serialized!");
    }

    private static ShorterJST shorterJSTMapper(JST jst)
    {
        ShorterJST mapped = new ShorterJST(jst.Kod_TERYT, jst.Województwo, jst.Powiat, jst.typ_JST,
            jst.nazwa_urzędu_JST, jst.miejscowość,
            Int32.Parse(jst.telefon_kierunkowy.ToString() + jst.telefon.ToString()));
        return mapped;
    }

}
